### PR TITLE
refactor and additions

### PR DESCRIPTION
**Pull request by CourtneyJa**
# Description

Fixes # (issue) - folders more organized to keep components together, jest testing components added, tables and seeds updated to fix pg auto inc error

## Type of change

Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [x] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: data/db_config.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 const knex = require('knex');
 
+
 const knexfile = require('../knexfile');
+require('dotenv').config();
 
 const env = process.env.NODE_ENV || 'development';
+
+// const env = 'development';
```
Comment (hisnameisjimmy) [inline data/db_config.js:9]: Remove `require('dotenv').config();`

Add a comment above note for only using 'development' environment and it working for testing

File: data/migrations/20200426223901_music.js [hunk 2]
```diff
@@ -13,7 +13,6 @@ exports.up = async function (knex, Promise) {
     song.string("title").notNullable();
     song.string("title_short").notNullable();
```
Comment (hisnameisjimmy) [inline data/migrations/20200426223901_music.js:14]: We also want to remove this

File: data/seeds/02_songs.js [hunk 3]
```diff
@@ -4,60 +4,46 @@ exports.seed = function (knex, Promise) {
     .then(function () {
       return knex("songs").insert([
         {
-          id: 1,
           deezer_id: "12",
           title: "Push Every Button",
           title_short: "Push Every Button",
```
Comment (hisnameisjimmy) [inline data/seeds/02_songs.js:9]: Remove title_short

File: pull_request_template.md [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
-Fixes # (issue)
+Fixes # (issue) - folders more organized to keep components together, jest testing components added, tables and seeds updated to fix pg auto inc error
 
 ## Type of change
 
 Please delete options that are not relevant.
 
-- [ ] Bug fix (non-breaking change which fixes an issue)
+- [x] Bug fix (non-breaking change which fixes an issue)
```
Comment (hisnameisjimmy) [inline pull_request_template.md:9]: We shouldn't be adjusting the template itself, only in the PR.

File: api/api_router.spec.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const request = require("supertest");
+const router = require("./api_router");
+
+describe("api_router", () => {
```
Comment (hisnameisjimmy) [inline api/api_router.spec.js:4]: Let's leave a comment that this doesn't run yet

File: data/db_config.js (hunk 1 above)
Comment (CourtneyJa) [reply data/db_config.js:7]: This has been completed with a note added

**Comment by CourtneyJa** [comment]
> See comments.
> 
> Also, let's manually test via postman all the existing routes to make sure nothing broke in the refactor.

Tested Login, Register, GET Songs & Videos

**Pull request merged**